### ISSUE TITLE
Clean-up, delete, and consolidate top-level files/directories in the repository

### ISSUE THREAD

**Issue by WillDaSilva** [meltano/meltano#6262]
The top-level of the Meltano repository is cluttered. This serves to hide our readme when one visits https://github.com/meltano/meltano. For that reason and more, we should probably delete the outdated/unused files and directories, and consolidate configs where possible.

To (potentially) delete:
- [x] `.gitlab/` once the migration to GitHub Actions is complete
- [x] `cloud/`
- [x] `license_finder/` if it is no longer used after the migration away from GitLab CI is complete. Even if we still need it, it probably shouldn't be at the top level. @kgpayne
- [ ] `.adr-dir` @pandemicsyn what is the point of this file? Does some tool we use depend on it?
- [x] `.dockerignore` if it is not doing anything. If I'm not mistaken the `.dockerignore` file needs to be at the root of the build context to have any effect, and as far as I know we don't use the top-level of the Meltano repository as a Docker build context.
- [x] `.env.dev`
- [x] `.env.example`
- [x] `.gitlab-ci.yml` once the migration to GitHub Actions is complete
- [ ] `.tmuxinator.yml`
- [x] `Caddyfile`
- [x] `Makefile_explain.sh`
- [ ] `alembic.ini` if Alembic is no longer needed
- [x] `docker-compose.yml`
- [x] `setup.cfg`

Aside from that, we can move some config into `pyproject.toml` (n.b. `pyproject.toml` aims to store the configs of more or less all Python dev tools):
- If we are willing to write [`pflake8`](https://github.com/csachs/pyproject-flake8) instead of `flake8`, then we can put the flake8 config (`.flake8`) in `pyproject.toml`
- [`pytest.ini` can be moved into `pyproject.toml`](https://docs.pytest.org/en/latest/reference/customize.html#pyproject-toml)
- [`.coveragerc` can be moved into `pyproject.toml`](https://coverage.readthedocs.io/en/latest/config.html)
  > Coverage.py will read settings from other usual configuration files if no other configuration file is used. It will automatically read from “setup.cfg” or “tox.ini” if they exist. In this case, the section names have “coverage:” prefixed, so the [run] options described below will be found in the [coverage:run] section of the file. If coverage.py is installed with the toml extra (pip install coverage[toml]), it will automatically read from “pyproject.toml”. Configuration must be within the [tool.coverage] section, for example, [tool.coverage.run].

@aaronsteers @tayloramurphy @pandemicsyn @edgarrmondragon

**Comment by tayloramurphy** [comment]
I'm in favor of this but assigning to Internal WIP as it's more on Engineering to decide the best organization.

**Comment by pandemicsyn** [comment]
> .adr-dir @pandemicsyn what is the point of this file? Does some tool we use depend on it?

@WillDaSilva yea it lets [adr-tools](https://github.com/npryce/adr-tools) know what dir our adr's live in. We could ommit it and just make a small note in https://docs.meltano.com/contribute/merge#architectural-decision-records

**Comment by WillDaSilva** [comment]
Once Poetry core version 1.1.0 has been released, we'll be able to move `build.py` into a sub-directory, e.g. `scripts/`, since they have fixed the bug where it must be at the top-level.

**Comment by WillDaSilva** [comment]
Implemented by:
- https://github.com/meltano/meltano/pull/6837
- https://github.com/meltano/meltano/pull/6838
- https://github.com/meltano/meltano/pull/6839
- https://github.com/meltano/meltano/pull/6842
- https://github.com/meltano/meltano/pull/6843
- https://github.com/meltano/meltano/pull/6847
